SumoLogic.Logging.Log4Net/BufferedSumoLogicAppender.cs

```diff
@@ -268,6 +269,12 @@ protected override void Append(LoggingEvent loggingEvent)
                 return;
             }
 
+            if(Layout == null)

```

Comment:
This will be checked/warned on every single log message? Could the default layout be set in the constructor?

Reply:
Would it make sense to throw a ConfigurationException in the constructor so that a consumer of the logger knows they are missing a vital configuration piece?

Reply:
My guess is that when config is XML-specified, Log4Net will need to construct the object in some kind of default state, then set properties after the fact via reflection. That's just a hunch, would be useful to test that theory.

What do other established appenders do? Another option might be to define a custom `MissingFormatException` instead of just nullrefing.

Reply:
it appears what you are supposed to do is override RequiresLayout

```csharp
		override protected bool RequiresLayout
		{
			get { return true; }
		}

```

I found this in Apache's log4net examples. I put the console example up [as a gist on github](https://gist.github.com/TerribleDev/d0182004c88aa748d58969edce5ca7b2#file-consoleappenderexample-cs-L180)

Reply:
I think then you are then guaranteed that it won't be null. I assume Log4Net throws some kind of human understandable error.